Core/src/org/sleuthkit/autopsy/corecomponents/DataContentViewerHTML.java

```diff
@@ -0,0 +1,293 @@
+ */
+@ServiceProvider(service = DataContentViewer.class, position = 6)
+public class DataContentViewerHTML extends javax.swing.JPanel implements DataContentViewer {
+
+    private AbstractFile dataSource;
+    private FileTypeDetector fileTypeDetector;
+    private Stage stage;
+    private WebView browser;
+    private JFXPanel jfxPanel;
+    private WebEngine webEngine;
+    private Scene scene;
+    private Button javaScriptButton;
+    
+    private final String enableJavaScriptTxt;
+    private final String disableJavaScriptTxt;
+    
+    private static final Logger logger = Logger.getLogger(DataContentViewerHTML.class.getName());
+    
+  
+    public DataContentViewerHTML() {
+        
+        try {
+            fileTypeDetector = new FileTypeDetector();
+        } catch (FileTypeDetectorInitException ex) {
+            Exceptions.printStackTrace(ex);
+        }
+        
+        initComponents();
+        resetComponents();
+        
+        enableJavaScriptTxt = NbBundle.getMessage(this.getClass(), 
+                "DataContentViewerHTML.enableJavaScriptBtnTxt");
+        disableJavaScriptTxt = NbBundle.getMessage(this.getClass(), 
+                "DataContentViewerHTML.disableJavaScriptBtnTxt");        
+        
+        
+        logger.log(Level.INFO, "Created HTMLView instance: " + this); //NON-NLS
+    }
+    
+    private void initComponents() {     
+        
+       jfxPanel = new JFXPanel();
+       setLayout(new BorderLayout());
+       add(jfxPanel, BorderLayout.CENTER);
+       createWebViewScene();     
+    }
+    
+    private void resetComponents() {
+        
+        //Disable JavaScript 
+        PlatformImpl.runLater(()-> {
+            webEngine.setJavaScriptEnabled(false);
+            javaScriptButton.setText(enableJavaScriptTxt);
+        });
+        
+        //Load an empty "page"
+        loadWebEngineContent("");
+    }
+    
+    /**
+     * Loads HTML content into the webEngine as long as the engine is not null and the
+     * content is not null. 
+     * @param content Not null. The content to load into the web engine.
+     */
+    private void loadWebEngineContent(String content) {
+        if(content != null && webEngine != null) {
+            PlatformImpl.runLater(() -> {
+                webEngine.loadContent(content);
+            });
+        }
+    }
+    
+    /**
+     * Loads an HTML file into the webEngine as long as the engine is not null and the
+     * content is not null.
+     * @param localFilePath Not null. The  local file path to the HTML file to be loaded.
+     */
+    private void loadWebEngineFileURL(String localFilePath) {
+        
+        if(localFilePath != null && webEngine != null) {
+            PlatformImpl.runLater(() -> {
+                webEngine.load("file://" + localFilePath);
+            });
+        }
+    }
+    
+    private void createWebViewScene() {
+        PlatformImpl.startup(() -> {
```

Comment:
see above about PlatformImpl vs Platform

Platform does not have a startup method, but I don't see why this couldn't be done with runLater